providers/sum/sum.go

```diff
@@ -0,0 +1,280 @@
+package sum
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/bmc-toolbox/bmclib/v2/providers"
+	"github.com/bmc-toolbox/common"
+	"github.com/bmc-toolbox/common/config"
+	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/registrar"
+)
+
+const (
+	// ProviderName for the provider implementation
+	ProviderName = "sum"
+	// ProviderProtocol for the provider implementation
+	ProviderProtocol = "sum"
+)
+
+var (
+	// Features implemented by SUM
+	Features = registrar.Features{
+		providers.FeatureGetBiosConfiguration,
+		providers.FeatureSetBiosConfiguration,
+		providers.FeatureSetBiosConfigurationFromFile,
+		providers.FeatureResetBiosConfiguration,
+	}
+)
+
+// Conn for Sum connection details
+type Exec struct {
+	SumPath  string
+	Log      logr.Logger
+	Host     string
+	Username string
+	Password string
+}
+
+// Option for setting optional Client values
+type Option func(*Exec)
+
+func WithSumPath(sumPath string) Option {
+	return func(c *Exec) {
+		c.SumPath = sumPath
+	}
+}
+
+func WithLogger(log logr.Logger) Option {
+	return func(c *Exec) {
+		c.Log = log
+	}
+}
+
+func New(host, user, pass string, opts ...Option) (*Exec, error) {
+	sum := &Exec{
+		Host:     host,
+		Username: user,
+		Password: pass,
+		Log:      logr.Discard(),
+	}
+
+	for _, opt := range opts {
+		opt(sum)
+	}
+
+	var err error
+
+	if sum.SumPath == "" {
+		sum.SumPath, err = exec.LookPath("sum")
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		if _, err = os.Stat(sum.SumPath); err != nil {
+			return nil, err
+		}
+	}
+
+	return sum, nil
+}
+
+// Open a connection to a BMC
+func (c *Exec) Open(ctx context.Context) (err error) {
+	return nil
+}
+
+// Close a connection to a BMC
+func (c *Exec) Close(ctx context.Context) (err error) {
+	return nil
+}
+
+func (c *Exec) Name() string {
+	return ProviderName
+}
+
+func (s *Exec) run(ctx context.Context, command string, additionalArgs ...string) (output string, err error) {
+	// TODO(splaspood) use a tmp file here (as sum supports) to read the password
+	sumArgs := []string{"-i", s.Host, "-u", s.Username, "-p", s.Password, "-c", command}
+	sumArgs = append(sumArgs, additionalArgs...)
+
+	s.Log.V(9).WithValues(
+		"sumArgs",
+		sumArgs,
+	).Info("Calling sum")
```

Comment:
suggest using Debug or Trace instead of Info

Reply:
I believe this logger uses that V(9) to define what level of 'verbosity' something is logged at?    Correct me if I'm wrong...   but there's no .Debug() available off that as-is.

Reply:
you're right there, ignore my suggestion